src/main/java/com/ldtteam/structurize/util/BlockUtils.java

```diff
@@ -220,6 +223,49 @@ else if (blockState.getBlock() == Blocks.BAMBOO_SAPLING)
         }
     }
 
+    /**
+     * For structure placement, check if two blocks are alike or if action has to be taken.
+     * @param blockState1 the first blockState.
+     * @param blockState2 the second blockState.
+     * @return true if nothing has to be done.
+     */
+    public static boolean areBlockStatesEqual(final BlockState blockState1, final BlockState blockState2)
+    {
+        if (blockState2 == null)
+        {
+            return true;
```

Comment:
equals returns true if the second is null? that seems weird/different from what I would expect it to return

Reply:
I'll add it for both, but if blockstate2 is null, this means that we're likely outside of the schematic in this call somehow, so we want to handle this smoothly.

Reply:
there is Wold#isOutsideBuildHeight or isOutsideY or however it's called, worth using that

Reply:
I mean it's when the blueprint returns null for a position.

Reply:
blueprint cant return null for pos, if it does then it's corrupted

Reply:
and how do you get outside of the schematic?

Reply:
It can, if out of range. (-1 pos for example)